File: 01 Two Pointers/Two Sum /Solution.java
import java.util.Scanner;
import java.util.Arrays;
import java.util.List;
import java.util.stream.Collectors;
import java.util.ArrayList;

public class Solution{

    private static List<String> split(String s){
        return s.isEmpty() ? List.of() : Arrays.asList(s.split(" "));
    }
 

    private static List<Integer> twoSum(List<Integer> a, int target){
        var res = new ArrayList<Integer>();
        var l = 0;
        var h = a.size() - 1;
        while( l < h ){
            var sum = a.get(l) + a.get(h);
            if( sum < target ){
                l++;
            } else if( sum > target ){
                h--;
            } else{
                res.add(l+1);
                res.add(h+1);
                return res;
            }
        }
        return res;
    }

    public static void main(String ...s){
        try(Scanner scan = new Scanner(System.in)){
            System.out.println("Enter the elements : ");
            List<Integer> list = split(scan.nextLine()).stream().map(Integer::parseInt).collect(Collectors.toList());
            System.out.println("Enter the target : ");
            int target = Integer.parseInt(scan.nextLine());
            System.out.println("Result : "+twoSum(list, target));
        }
    }
}

File: 01 Two Pointers/Two Sum /solution.py
#!/usr/bin/env python3

def two_sum(a : list[int], target: int)->list[int]:
    l = 0
    h = len(a) - 1
    while( l < h ):
        sum = a[l] + a[h]
        if sum < target:
            l += 1
        elif sum > target:
            h -= 1
        else:
            return [l+1,h+1]
    return []            

if __name__ == "__main__":
    print("Enter the Elements : ")
    list = [int(x) for x in input().split()]
    print("Enter the target : ")
    target = int(input())
    print("Result : ", two_sum(list,target))
